hazelcast/src/test/java/com/hazelcast/internal/memory/impl/HeapMemoryManager.java

```diff
@@ -91,12 +96,36 @@ public long allocate(long size) {
 
         @Override
         public long reallocate(long address, long currentSize, long newSize) {
-            throw new UnsupportedOperationException();
+            assertFitsInt(address);
+            assertFitsInt(currentSize);
+            assertFitsInt(newSize);
+            validateBlock(address, currentSize);
+            final long sizeDelta = newSize - currentSize;
+            final long newAddress;
+            if (address == lastAllocatedAddress) {
+                heapTop += sizeDelta;
+                newAddress = address;
```

Comment:
update old size with new size. address (key) is still same but size (value) is changed.